cloud/services/inboundnatrules/inboundnatrules_test.go

```diff
@@ -325,3 +341,201 @@ func TestDeleteNetworkInterface(t *testing.T) {
 		})
 	}
 }
+
+func TestNatRuleExists(t *testing.T) {
+	testcases := []struct {
+		name           string
+		ruleName       string
+		existingRules  []network.InboundNatRule
+		expectedResult bool
+		expectedPorts  map[int32]struct{}
+		expect         func(s *mock_inboundnatrules.MockInboundNatScopeMockRecorder,
+			m *mock_inboundnatrules.MockClientMockRecorder, mLoadBalancer *mock_loadbalancers.MockClientMockRecorder)
+	}{
+		{
+			name:     "Rule exists",
+			ruleName: "my-rule",
+			existingRules: []network.InboundNatRule{
+				{
+					InboundNatRulePropertiesFormat: &network.InboundNatRulePropertiesFormat{
```

Comment:
copying this test case and reversing the order should catch the other comment I just dropped

Reply:
it won't, we only care about the content of port if the rule doesn't already exists https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/893/files/b185781e9547bd6c9ba3b0a87ecf13d23ca6b637#diff-ecbb14ef6845369e03315e8fe557892eR438

Reply:
yep sorry